app/controllers/posts_controller.rb

```diff
@@ -24,7 +26,8 @@ def create
 
   def show
     set_user
-    @post = Post.find(params[:id])
+    # @post = Post.find(params[:id])
+    @post = Post.includes(:author).find_by(author_id: params[:user_id], id: params[:id])
```

Comment:
- [ ]  Kindly remove this commented code in order for your codebase to be cleaner. ℹ️
**N.B. This is also applicable to line 13 in post_index_spec.rb file.**